src/bp/lang-server/service/index.ts

```diff
@@ -45,75 +54,172 @@ export default class LanguageService {
     return this._ready
   }
 
-  listModels(): AvailableModel[] {
-    return _.values(this._models)
+  listFastTextModels(): AvailableModel[] {
+    const allModels = _.values(this._models)
+    return allModels.map(m => m.fastTextModel)
   }
 
   private _discoverModels() {
     if (!fs.existsSync(this.langDir)) {
-      return []
+      return
     }
 
     const files = fs.readdirSync(this.langDir)
-    return files.forEach(f => {
+
+    type modelFileInfo = {
+      domain: string
+      langCode: string
+      file: string
+      dim?: number
+    }
+    const filesInfo: modelFileInfo[] = []
+
+    files.forEach(f => {
       // Examples:  "scope.en.300.bin" "bp.fr.150.bin"
-      const regex = /(\w+)\.(\w+)\.(\d+)\.bin/i
-      const match = f.match(regex)
-      if (!match) {
+      const fastTextModelsRegex = /(\w+)\.(\w+)\.(\d+)\.bin/i
+      const fastTextModelsMatch = f.match(fastTextModelsRegex)
+
+      // Examples: "scope.en.bpe.model" "bp.en.bpe.model"
+      const bpeModelsRegex = /(\w+)\.(\w+)\.bpe\.model/i
+      const bpeModelsMatch = f.match(bpeModelsRegex)
+
+      if (!!fastTextModelsMatch) {
+        const [__, domain, langCode, dim] = fastTextModelsMatch
+        filesInfo.push({ domain, langCode, dim: Number(dim), file: f })
         return
       }
+      if (!!bpeModelsMatch) {
+        const [__, domain, langCode, _] = bpeModelsMatch
+        filesInfo.push({ domain, langCode, file: f })
+      }
+    })
+
+    const modelGroups = _.groupBy(filesInfo, x => [x.domain, x.langCode])
 
-      const [__, domain, langCode, dim] = match
+    _.forEach(modelGroups, v => {
```

Comment:
Try to be consistent with your method calls, you used the native forEach in the previous one, and lodash's one here. Also, why v ? Shouldn't "group" or "model" be more meaningful?